src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs

```diff
@@ -1004,9 +1004,10 @@ public sealed class EntryPointGraph
         private readonly RunContext _context;
         private readonly List<EntryPointNode> _nodes;
 
-        public EntryPointGraph(IHostEnvironment env, ComponentCatalog catalog, JArray nodes)
+        public EntryPointGraph(IHostEnvironment env, JArray nodes)
         {
             Contracts.CheckValue(env, nameof(env));
+            var catalog = env.ComponentCatalog;
```

Comment:
> var catalog = env.ComponentCatalog; [](start = 11, length = 36)

it only get used as parameter in ValidateNodes function, and we pass _host to it which also has catalog, so i would rather remove it from here and just got catalog inside ValidateNodes method.